packages/engine/lib/orchestrator/src/experiment.rs

```diff
@@ -95,6 +95,25 @@ pub struct ExperimentConfig {
     clap(global = true, short = 'w', long, default_value_t = num_cpus::get(), validator = at_least_one, env = "HASH_WORKERS")
     )]
     pub num_workers: usize,
+
+    #[cfg_attr(feature = "clap", clap(global = true, long, default_value_t = 0))]
+    /// Initial size of the V8 heap.
+    ///
+    /// Setting this value allows to avoid garbage collection while the heap is small enough.
+    pub v8_initial_heap_constraint: usize,
```

Comment:
Can we use `Option<usize>` instead? Instead of providing the weird amount "0", we better fallback to default.

Reply:
We can I think it would only move the problem elsewhere as there is no way to tell V8 to use the default if we set the max heap constraint.

Reply:
It's different if the problem is user-facing or it's an implementation detail. When initializing v8, we can check, if any constraint is set and simply `.unwrap_or(0)`.

Reply:
It's different if the problem is user-facing or it's an implementation detail. When initializing v8, we can check, if any constraint is set and simply `.unwrap_or(0)`.

Reply:
How will users find this value?
If we have to add it to the documentation, I'd rather have clap do it.

Or we could not add this variable to the cli at all in the light of what it does.

Reply:
Why do user need to find the default value? If both values are not set, we can simply use the v8 default values.

Reply:
Use `Option` in https://github.com/hashintel/hash/pull/446/commits/39550930d1416cf96fc927fb51721b2c0c7c0f77